### ISSUE TITLE
Nodemon not restarting when using node-java

### ISSUE THREAD

**Issue by dannyn-mirth** [remy/nodemon#573]
I posted this issue over at the node-java project: https://github.com/joeferner/node-java/issues/226

In a nutshell, when using node-java, nodemon doesn't start the server after it detects file changes. I thought this was a node-java issue but when I tried out supervisor, it was able to restart the server as it should.

**Comment by ezrepotein** [comment]
+1

**Comment by remy** [comment]
Firstly, I've no clue what `node-java` is, so you may need to investigate
and fix yourselves. More importantly, are you using the latest nodemon?
Have you also tried in legacy mode?

**Comment by tomaskulich** [comment]
+1

happens with `nodemon` 1.8.1

Some clues what might be happening: Using `node-java` (and `tika` in my case), the app does respond to `SIGUSR2` signal correctly (I think `nodemon` somehow assumes / sets that sending `SIGUSR2` should kill the app?).

I managed to correct the behavior in my case by adding:

```javascript
process.once('SIGUSR2', function() {
  process.kill(process.pid, 'SIGUSR2')
})
```

which is quite funny since I'd assume that the script should have no effect on anything.

**Comment by ArturFormella** [comment]
+1

**Comment by naxmefy** [comment]
seems that SIGUSR2 is never called when node-java is there

**Comment by camwes** [comment]
@dannyn-mirth I would  suggest that this is closed by #1061. As @tomaskulich correctly pointed out the issue is with the signal being passed to node-java. If you use `SIGINT` instead the application should terminate normally.

**Comment by naxmefy** [comment]
@dannyn-mirth i switched to pm2 (with pm2-dev) for projects with node-java - works absolutly fine - tried every signal from nodemon without luck... cuz node-java seems to need the signal multiple times or sth like dis cya
